File: ProjectEuler Java/ProjectEuler010.java
import java.util.ArrayList;

/**
 * Summation of primes
 * Problem 10
 * 
 * @author Oskar Mendel
 */
public class ProjectEuler010 {

	public static void main(String[] args) {
		ArrayList<Integer> list = genPrimes(2000000);
		long sum = 0;
		for (Integer x : list) {
			sum += x;
		}
		
		System.out.println(sum);
	}

	public static ArrayList<Integer> genPrimes(int n) {
		ArrayList<Integer> list = new ArrayList<Integer>();
		for (int i = 2; i <= n; i++) {
			if (isPrime(i)) {
				list.add(i);
			}
		}
		
		return list;
	}
	
	public static boolean isPrime(int n) {
		if (n == 1) {
			return false;
		}
		
		if (n == 2) {
			return true;
		}
		
		double d = Math.sqrt(n);
		
		for (int i = 2; i <= d; i++) {
			if (n % i == 0) {
				return false;
			}
		}
		
		return true;
	}
}

File: ProjectEuler C++/ProjectEuler010.cpp
#include <iostream>
#include <math.h>
#include <vector>

/*
 * Summation of primes
 * Problem 10
 *
 * The sum of the primes below 10 is 2 + 3 + 5 + 7 = 17.
 *
 * Find the sum of all the primes below two million.
 */

bool isPrime(int num)
{
	if (num == 1 || num < 1)
	{
		return false;
	}

	if (num == 2)
	{
		return true;
	}

	double d = sqrt(num);

	for (int i = 2; i <= d; i++)
	{
		if (num % i == 0)
		{
			return false;
		}
	}

	return true;
}

int main(int argc, char** argv)
{
	long long sum = 0;

	for (int i = 2; i < 2000000; i++)
	{
		if (isPrime(i))
		{
			sum += i;
		}
	}
	
	std::cout << sum << std::endl;
	return 0;
}